File: apps/server/src/routes/posts.ts
import { createPost, getPosts, getPostsBySearchTerm, likePost } from "../controllers/posts.js";

import type { FastifyPluginCallback } from "fastify";

const postRoutes: FastifyPluginCallback = async fastify => {
  fastify.get("/posts", getPosts);
  fastify.post("/posts", createPost);
  fastify.get("/posts/search", getPostsBySearchTerm);
  fastify.patch("/posts/:id/like", likePost);
};

export default postRoutes;

File: apps/server/routes/posts.go
package routes

import (
	post_controllers "tea-share/controllers/posts"

	"github.com/fasthttp/router"
)

func RegisterPostRoutes(router *router.Router) {
	router.GET("/posts", post_controllers.GetPosts)
	router.GET("/posts/search", post_controllers.GetPostsBySearchTerm)
	router.POST("/posts", post_controllers.CreatePost)
	router.PATCH("/posts/{id}/like", post_controllers.LikePost)
}
